source/common/router/router.cc

```diff
@@ -83,8 +83,9 @@ void FilterUtility::setUpstreamScheme(Http::RequestHeaderMap& headers, bool down
     if (Http::HeaderUtility::schemeIsValid(headers.getSchemeValue())) {
       return;
     }
-    if (Http::HeaderUtility::schemeIsValid(headers.getForwardedProtoValue())) {
-      headers.setScheme(headers.getForwardedProtoValue());
+    absl::string_view xfp = headers.getXForwardedProtoValue();
+    if (Http::HeaderUtility::schemeIsValid(xfp)) {
+      headers.setScheme(xfp);
```

Comment:
This is hurting my brain (should have done this earlier in the day). How come this case isn't using the scheme header if we are preserving downstream scheme? More comments?